packages/hoprd/src/admin.ts

```diff
@@ -25,6 +25,35 @@ export class AdminServer {
 
   constructor(private logs: LogStream, private host: string, private port: number) {}
 
+  authenticate(req): boolean {
+    if (process.env.API_TOKEN === undefined) {
+      this.logs.log('ws connection authenticated: no token set')
+      return true
+    }
+    const cookies = this.parseCookies(req.headers.cookie)
+    if (cookies['x-auth-token'] !== process.env.API_TOKEN) {
+      this.logs.log('ws connection authentication failed: wrong token')
+      return false
+    }
+    this.logs.log('ws connection authenticated with token')
+    return true
+  }
+
+  parseCookies(rawCookies?: string): { [key: string]: string } {
+    const parsedCookies = {}
```

Comment:
why not use a library that can do this?

Reply:
Because default node.js `http` module doesn't allow any middleware like `express.js`. Wanna switch?